src/atomWithPending.ts

```diff
@@ -0,0 +1,5 @@
+import { atom } from 'jotai';
+import { pendingPromise } from './constants';
+
+export const atomWithPending = <Value>() =>
+  atom(pendingPromise as unknown as Value);
```

Comment:
The proper typing should be:
```ts
  atom(pendingPromise as Promise<never> | Value);
```
Does it work?

Reply:
Yes, but the setter is mis-typed:

<img width="522" alt="image" src="https://user-images.githubusercontent.com/11076888/169023169-27c3a3ec-5597-462a-a888-30faea9f2494.png">

We can specify the exact separated types for the getter and setter, but won't that make it harder to understand?

Reply:
Ah, you are right. So maybe this?
```ts
atom<Promise<never> | Value, SetStateAction<Value>>(pendingPromise);
```

In the official jotai docs, we could show practical type assertion, but as a library, it should be better to have precise types.

Reply:
Nope:
<img width="553" alt="image" src="https://user-images.githubusercontent.com/11076888/169024801-051d62d6-871a-4ffc-a69b-864edf5ece27.png">

Reply:
If typed correctly, it is:

```ts
type PendingAtom<Value> = Atom<Promise<never> | Value> &
  Omit<WritableAtom<Value, SetStateAction<Value>>, 'read'>;

export const atomWithPending = <Value>() =>
  atom(pendingPromise) as unknown as PendingAtom<Value>;
```

Reply:
oh, no! yeah, that signature is unsupported. 😓 

This would be the only solution:
```ts
atom(pendingPromise as Promise<never> | Value) as WritableAtom<Promise<never> | Value, SetStateAction<Value>>;
```

https://codesandbox.io/s/youthful-architecture-um22mv?file=/src/App.tsx:84-233

Oookay, I agree this can be too much. I'm fine with your original suggestion.